include/git2/remote.h

```diff
@@ -474,6 +474,16 @@ typedef int GIT_CALLBACK(git_push_negotiation)(const git_push_update **updates,
  */
 typedef int GIT_CALLBACK(git_push_update_reference_cb)(const char *refname, const char *status, void *data);
 
+/**
+ * Callback to resolve URLs before connecting to remote
+ *
+ * @param url The URL to resolve
+ * @param direction GIT_DIRECTION_FETCH or GIT_DIRECTION_PUSH
+ * @param payload The callback payload
+ * @return The resolved URL
+ */
+typedef const char *GIT_CALLBACK(git_resolve_url_cb)(const char *url, int direction, void *payload);
```

Comment:
This should return an `int` like other callbacks and instead use an out-parameter for the resolved URL. Furthermore, the out-parameter should be non-const. After having it resolved, the caller of `git_resolve_url_cb` should be responsible for freeing it again.